lib/workflows/utils.js

```diff
@@ -34,16 +34,35 @@ const check = (config) => {
     return 'Workflow steps must be an array';
   }
 
-  config.steps.forEach(step => {
-    if (!step.name) {
+  if (config.steps.length < 1) {
+    return 'Workflow must have at least one step';
+  }
+
+  const test = config.steps.map(step => {
+    if (!step.name || step.name === '') {
       return 'Step must have a name';
     }
+    if (typeof step.name !== 'string') {
+      return 'Step name must be a string';
+    }
     if (step.self && typeof step.self !== 'boolean') {
-      return 'Whether a step is self-publish must be a boolean';
+      return 'Self-publish must be a boolean';
+    }
+
+    return true;
+  }).filter((res) => {
+    if (res !== true) {
+      return res;
```

Comment:
Totally understand how filter works. I am determining what I do _not_ want in this array - ie, all the `true` ones. I am using `map` to test all of the steps, then I use `filter` to...wait for it...*_filter_* out the `true` ones, leaving me with either an array of strings (of which I return the first) or an empty array, wherein I return the truest true that was ever true.

The return of the string, although functional, is confusing though. I'll switch it to true.

Here is a codepen breaking it down: http://codepen.io/scottnath/pen/EyRaWy?editors=1111